File: server/models/room.ts
import {Message} from "./message";

export class Room {
    static MAXIMUM_USERS: number = 2;

    messages: Message[] = [];
    started: boolean = false;
    
    constructor(public id: string, public users: string[]) {
    }
    
    isFull(): boolean {
        return this.users.length >= Room.MAXIMUM_USERS;
    }
}

File: web/lib/models/room.dart
import 'package:flutter/material.dart';
import 'package:web/models/message.dart';

class Room {
  final String id;
  final List<String> users;
  final List<Message> messages;
  final bool started;

  Room(
      {@required this.id,
      @required this.users,
      @required this.messages,
      @required this.started});

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
        id: json['id'],
        users: new List<dynamic>.from(json['users']).map<String>((e) => e['userId']).toList(),
        messages: new List<dynamic>.from(json['messages'])
            .map((e) => Message.fromJson(e)).toList(),
        started: json['started']);
  }
}

